bazel/src/main/java/com/google/api/codegen/bazel/BazelBuildFileView.java

```diff
@@ -81,15 +81,24 @@
 
     Set<String> javaTests = new TreeSet<>();
     for (String service : bp.getServices()) {
-      String javaPackage = bp.getLangGapicPackages().get("java");
+      // Prioritize the language override in gapic.yaml if it is present.
+      // New APIs (circa 2020) should rely on the protobuf options instead.
+      String javaPackage =
+          bp.getLangGapicPackages().containsKey("java")
+              ? bp.getLangGapicPackages().get("java")
+              : bp.getLangProtoPackages().get("java");
       if (javaPackage == null) {
         continue;
       }
 
       String actualService =
-          bp.getLangGapicNameOverrides()
-              .get("java")
-              .getOrDefault(bp.getProtoPackage() + "." + service, service);
+          bp.getLangGapicNameOverrides().containsKey("java")
```

Comment:
I've seen recently the test targets for java test rules generated with empty test class list. Looks like this has potential of fixing that, right? If yes, then it is awesome!